src/core/expression/qgsexpression.cpp

```diff
@@ -432,11 +432,10 @@ void QgsExpression::setAreaUnits( QgsUnitTypes::AreaUnit unit )
 QString QgsExpression::replaceExpressionText( const QString &action, const QgsExpressionContext *context, const QgsDistanceArea *distanceArea )
 {
   QString expr_action;
-
+  static const QRegularExpression sRegEx{ QStringLiteral( "\\[%(.*?)%\\]" ),  QRegularExpression::MultilineOption | QRegularExpression::DotMatchesEverythingOption };
```

Comment:
Needs to be thread local - we can't use static regexs as they aren't thread safe